--- a/src/mktemp.rs
+++ b/src/mktemp.rs
@@ -550,7 +550,7 @@
     // The directory is created with these permission at creation time, using mkdir(3) syscall.
     // This is not relevant on Windows systems. See: https://docs.rs/tempfile/latest/tempfile/#security
     // `fs` is not imported on Windows anyways.
-    #[cfg(not(windows))]
+    #[cfg(unix)]
     builder.permissions(fs::Permissions::from_mode(0o700));

     match builder.tempdir_in(dir) {
