--- a/src/ln.rs
+++ b/src/ln.rs
@@ -22,6 +22,9 @@
 #[cfg(any(unix, target_os = "redox"))]
 use std::os::unix::fs::symlink;
 #[cfg(windows)]
 use std::os::windows::fs::{symlink_dir, symlink_file};
+#[cfg(target_os = "wasi")]
+#[allow(deprecated)]
+use std::fs::soft_link as symlink;
 use std::path::{Path, PathBuf};
 use uucore::backup_control::{self, BackupMode};
 use uucore::fs::{MissingHandling, ResolveMode, canonicalize};
