diff --git a/src/main/http/CMakeLists.txt b/src/main/http/CMakeLists.txt
index 2dccc81..a58bd58 100644
--- a/src/main/http/CMakeLists.txt
+++ b/src/main/http/CMakeLists.txt
@@ -1,3 +1,4 @@
+if(NOT DISABLE_EXTENSION_LOAD)
 include_directories(../../../third_party/httplib)
 
 # work-around for httplib
@@ -10,3 +11,4 @@ add_library_unity(duckdb_common_http OBJECT http_util.cpp)
 set(ALL_OBJECT_FILES
     ${ALL_OBJECT_FILES} $<TARGET_OBJECTS:duckdb_common_http>
     PARENT_SCOPE)
+endif()
