--- a/src/sys/mod.rs
+++ b/src/sys/mod.rs
@@ -12,6 +12,7 @@
         pub(crate) use std::os::windows::io::AsHandle as AsOpenFile;
     } else {
         mod unsupported;
-        pub use unsupported;
+        pub use unsupported::*;
+        pub(crate) use std::os::fd::AsRawFd as AsOpenFile;
     }
 }
--- a/src/sys/unsupported/read_guard.rs
+++ b/src/sys/unsupported/read_guard.rs
@@ -1,5 +1,5 @@
 use std::ops;
-use std::os::unix::io::AsRawFd;
+use std::os::fd::AsRawFd;
 
 use super::RwLock;
 
--- a/src/sys/unsupported/rw_lock.rs
+++ b/src/sys/unsupported/rw_lock.rs
@@ -1,5 +1,5 @@
 use std::io::{self, Error, ErrorKind};
-use std::os::unix::io::AsRawFd;
+use std::os::fd::AsRawFd;
 
 use super::{RwLockReadGuard, RwLockWriteGuard};
 
--- a/src/sys/unsupported/write_guard.rs
+++ b/src/sys/unsupported/write_guard.rs
@@ -1,5 +1,5 @@
 use std::ops;
-use std::os::unix::io::AsRawFd;
+use std::os::fd::AsRawFd;
 
 use super::RwLock;
 
